lib/_http_client.js

```diff
@@ -840,7 +842,7 @@ function _deferToConnect(method, arguments_, cb) {
 
   const callSocketMethod = () => {
     if (method)
-      this.socket[method].apply(this.socket, arguments_);
+      FunctionPrototypeApply(this.socket[method], this.socket, arguments_);
```

Comment:
Instead of `FunctionPrototypeApply`, this should use `ReflectApply`, which doesn’t go through the `uncurryThis` closure:
```suggestion
      ReflectApply(this.socket[method], this.socket, arguments_);
```